OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -58,6 +58,29 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						break;
 					}
 
+				case "PlaceBeacon":
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client != null)
+						{
+							var player = world.FindPlayerByClient(client);
+							if (player == null)
+							{
+								if (orderManager.LocalClient.IsObserver)
+								{
+									Game.AddChatLine(Color.White, ServerChatName, "{0} placed a beacon".F(client.Name));
+									player = world.Players[0];
```

Comment:
Yes please, that also looks like what the [ObserverShroudSelectorLogic](https://github.com/OpenRA/OpenRA/blob/e9926fdc2856ba6b3fb8a18883950a5a5f7aa0e5/OpenRA.Mods.Common/Widgets/Logic/Ingame/ObserverShroudSelectorLogic.cs#L66) is using.